presto-native-execution/presto_cpp/main/PeriodicTaskManager.h

```diff
@@ -68,10 +68,12 @@ class PeriodicTaskManager {
         func, std::chrono::microseconds{periodMicros}, taskName);
   }
 
-  /// Add a task to run once.
+  /// Add a task to run once. Before adding, cancels the any task that has same
```

Comment:
```
/// Add a task to run once.
void addTaskOnce(TFunc&& func, size_t periodMicros, const std::string& taskName) {
  // NOTE: before adding the function, we need to cancel any function with the same name. It is because folly::FunctionScheduler only removes a run-once function after its execution internally.
  ...
```
The note is specific to folly::FunctionScheduler implementation and other executor might not has such limitation.